### DISCUSSION TITLE
How could I return default value with response_model

### DISCUSSION THREAD

**Discussion by skk294** [fastapi/fastapi#8816] [Questions] [answered]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/samuelcolvin/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from pydantic import BaseModel
from typing import List, Optional
from fastapi import FastAPI

app = FastAPI()

class Users(BaseModel):
    name: Optional[str] = 'No Username'

@app.get("/", response_model=List[Users])
async def get_users():
    users = [{'id':1, 'name':'abc'},{'id':2, 'name':None}]
    return users
```

### Description

I want to return the imformation of all users. If an user didn't set his name(None), then return 'No username'.

### Operating System

Linux

### FastAPI Version

0.66.1

### Python Version

3.9.5

**Comment by Honda-a** [comment]
if you try this 
```
from pydantic import BaseModel
from typing import List, Optional
from fastapi import FastAPI

app = FastAPI()

class Users(BaseModel):
    name: Optional[str] = 'No Username'

@app.get("/", response_model=List[Users])
async def get_users():
    users = [{'id':1, 'name':'abc'},{'id':2}]
    return users
```

it will work 
but in your code you are setting the name to `None` so it will return None

**Comment by panla** [comment]
```python
from pydantic import Field

name: Optional[str] = Field('No username')
```

**Comment by skk294** [comment]
> if you try this
> 
> ```
> from pydantic import BaseModel
> from typing import List, Optional
> from fastapi import FastAPI
> 
> app = FastAPI()
> 
> class Users(BaseModel):
>     name: Optional[str] = 'No Username'
> 
> @app.get("/", response_model=List[Users])
> async def get_users():
>     users = [{'id':1, 'name':'abc'},{'id':2}]
>     return users
> ```
> 
> it will work
> but in your code you are setting the name to `None` so it will return None

Thank you for answering my question. I set it to None because i'm using tortoise-orm which will return like this if I didn't set the name(null=True) of user

**Comment by Honda-a** [comment, marked as answer]
I see then in that case you need to set your default value in the Model and use `pydantic_model_creator` it would be something like this

```
from tortoise.models import Model
from tortoise import fields
from tortoise.contrib.pydantic import pydantic_model_creator
class ModelUsers(Model):
    id = fields.IntField(pk=True)
    name = fields.TextField(null=True, default="No Username")

Users = pydantic_model_creator(ModelUsers, name="Users")

@app.get("/", response_model=List[Users])
async def get_users():
    users = await pydantic_model_creator.from_queryset(ModelUsers.all())
    return users
```

**Comment by skk294** [comment]
> I see then in that case you need to set your default value in the Model and use `pydantic_model_creator` it would be something like this
> 
> ```
> from tortoise.models import Model
> from tortoise import fields
> from tortoise.contrib.pydantic import pydantic_model_creator
> class ModelUsers(Model):
>     id = fields.IntField(pk=True)
>     name = fields.TextField(null=True, default="No Username")
> 
> Users = pydantic_model_creator(ModelUsers, name="Users")
> 
> @app.get("/", response_model=List[Users])
> async def get_users():
>     users = await pydantic_model_creator.from_queryset(ModelUsers.all())
>     return users
> ```

Thank you. ^_^
